cpp/ql/src/Security/CWE/CWE-611/XXE.ql

```diff
@@ -164,7 +164,17 @@ class ParseFunction extends Function {
 }
 
 /**
- * Configuration for tracking XML objects and their states.
+ * The `createLSParser` function that returns a newly created `LSParser` object.
+ */
+class CreateLSParser extends Function {
+  CreateLSParser() {
+    this.hasName("createLSParser") and
+    this.getType().(PointerType).getBaseType().getName() = "DOMLSParser" // returns a `DOMLSParser *`.
```

Comment:
This should make it handle typedefs as well.
```suggestion
    this.getUnderlyingType().(PointerType).getBaseType().getName() = "DOMLSParser" // returns a `DOMLSParser *`.
```

Reply:
Oops. I think I gave you bad advice. The predicate is called `getUnspecifiedType` (which explains the consistency error reported by ql-for-ql).